extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/input/table/TableInputSpecSlicer.java

```diff
@@ -169,9 +243,40 @@ public RichSegmentDescriptor toRichSegmentDescriptor()
           segment.getInterval(),
           interval,
           segment.getVersion(),
-          segment.getShardSpec().getPartitionNum(),
-          segment instanceof DataSegmentWithLocation ? ((DataSegmentWithLocation) segment).getServers() : null
+          segment.getShardSpec().getPartitionNum()
       );
     }
+
+    @Override
+    public long getWeight()
+    {
+      return segment.getSize();
+    }
+  }
+
+  private static class DataServerRequest implements WeightedInputInstance
+  {
+    private static final long DATA_SERVER_WEIGHT_ESTIMATION = 5000L;
+    private final List<DataSegmentWithInterval> segments;
+    private final DruidServerMetadata serverMetadata;
+
+    public DataServerRequest(DruidServerMetadata serverMetadata, List<DataSegmentWithInterval> segments)
+    {
+      this.segments = Preconditions.checkNotNull(segments, "segments");
+      this.serverMetadata = Preconditions.checkNotNull(serverMetadata, "server");
+    }
+
+    @Override
+    public long getWeight()
+    {
+      return segments.size() * DATA_SERVER_WEIGHT_ESTIMATION;
```

Comment:
We are estimating the number of rows to be 5000 for realtime segments rite ?

Reply:
Yes, added a comment